### PR TITLE
Rework nRF Connect documentation

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
Enhance documentation of Nordic nRF Connect platform and examples.

 #### Summary of Changes
- Add more general guides on nRF Connect platform integration in CHIP, example configuration and CLI usage
- Rework README of all nRF Connect examples:
  - Describe capabilities of the examples in the Overview section
  - Add Requirements section which more precisely describes hardware and software requirements (prepare for support for other boards besides nRF52840 DK).
  - Try to make all described procedures less ambiguous.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by mspang** [comment]
CLA needs signing before this will go out to all approvers.

**Review by rwalker-apple — APPROVED** [review]

File: examples/lighting-app/nrfconnect/README.md [hunk 1]
```diff
@@ -1,302 +1,344 @@
...
+    -   [nRF Command Line Tools](https://www.nordicsemi.com/Software-and-Tools/Development-Tools/nRF-Command-Line-Tools)
+    -   [GN meta-build system](https://gn.googlesource.com/gn/)
 
-Download and install [GN meta-build system](https://gn.googlesource.com/gn/).
+2.  Depending on whether you have the nRF Connect SDK installed:
 
-Make sure that you source the following file:
+    -   Follow the
+        [guide](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_assistant.html#)
+        in the nRF Connect SDK documentation to install the nRF Connect SDK
```
Comment (bzbarsky-apple) [inline examples/lighting-app/nrfconnect/README.md:230]: So this points to using the gui tool to install the SDK, right?  It'd be nice if there were just pointers here to the right things to download, which can be derived, with some work, from the links that used to be here...

**Review by saurabhst — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: examples/lighting-app/nrfconnect/README.md (hunk 1 above)
Comment (Damian-Nordic) [reply examples/lighting-app/nrfconnect/README.md:230]: Have you tried the GUI tool (Getting Started Assistant)? Did you find any issues using it? We considered several options for this section. The article which used to be linked here, which describes manual steps for the setup, was unclear for some devs as it refers to other articles in the Zephyr documentation etc. and it is quite easy to get lost. Listing the steps in this README directly would require us to cover multiple operating systems and not to assume that a user has any software installed. We thought Getting Started Assistant might provide less error-prone solution for setting up the environment as it verifies all steps made so far. But let me know your thoughts on that so we may do another round of the documentation improvement :)

Comment (bzbarsky-apple) [reply examples/lighting-app/nrfconnect/README.md:230]: I haven't yet, but in the past I've had my share of trouble with some tools when I needed to run a build on a system I only had console access to....

**Pull request merged**